### PR TITLE
User Manual for Autojudge

### LINKED ISSUE #79: Design Training Manual
We need to design a training manual for people interested in using this software.

I’m adding a list of points to be covered (should be expanded):

- [x] Create Contest
- [x] Add posters for the contest
- [x] Add participants for the contest
- [x] Update dates
- [x] Add Problem for the contest
- [x] Edit Problem
- [x] Delete Problem
- [x] Delete contest
- [x] Add test cases
- [x] Delete test cases (is this there)
- [x] Make submission
- [x] View all submissions and the statuses
- [x] Add comment
- [x] Update poster score
- [x] Download submissions
- [x] Download CSV of submission scores

### PR DESCRIPTION
**Pull request by vbsinha**
Resolves #79.

### CODE REVIEW

File: docs/source/manual.rst [hunk 1]
```diff
@@ -0,0 +1,33 @@
+User Manual for ``autojudge``
+=============================
+
+Creating your account / Logging in
+----------------------------------
+
+On the home page, click ``LOG IN`` (see top right corner).
+Login via your IITH Google account.
```
Comment (vishwakftw) [inline docs/source/manual.rst]: Worth specifying that this uses OAuth.

File: docs/source/manual.rst [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+Basic Ideas for User
```
Comment (vishwakftw) [inline docs/source/manual.rst]: Could benefit from a better title.

“Key abstractions of autojudge” or sth like that.

File: docs/source/manual.rst [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+--------------------
+
+The judge uses a system of ``Contest`` s.
+A ``Contest`` consists of a set of ``Problem`` s.
+
+A user can be a ``Poster``, a ``Participant`` or neither for a ``Contest``.
+For each ``Contest`` a user will hold one and only one position.
+
+The user who creats a ``Contest`` becomes the ``Poster`` for the ``Contest``.
```
Comment (vishwakftw) [inline docs/source/manual.rst]: ... for the Contest by default.

Comment (vishwakftw) [reply docs/source/manual.rst]: Nit: who creates

File: docs/source/manual.rst [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+The ``Poster`` can add more ``Poster`` s, who would help in creating the problems.
```
Comment (vishwakftw) [inline docs/source/manual.rst]: Not restricted to just creating problems ;-).

File: docs/source/manual.rst [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+These ``Poster`` s together write a set of ``Problem`` s for the ``Contest``.
```
Comment (vishwakftw) [inline docs/source/manual.rst]: Again, this too.

File: docs/source/manual.rst [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+The ``Poster`` can leave the ``Contest`` open for all users to particpate, in which case it is called a ``Public Contest``.
+Or the ``Poster`` may specify a set of ``Participant`` s. In this case only these users can particpate in the ``Contest``. Such ``Contest`` s are also called ``Private Contest`` s.
```
Comment (vishwakftw) [inline docs/source/manual.rst]: Starting a sentence with Or seems weird. Maybe add it to the previous line as continuation.

File: docs/source/manual.rst [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+Note that no ``Poster`` can ever participate in a contest.
```
Comment (vishwakftw) [inline docs/source/manual.rst]: Why? (I know, it was rhetorical)

Perhaps worth mentioning the capabilities and incapabilities of Posters and Participants.

File: docs/source/manual.rst [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+If the ``Contest`` is ``Public`` every user is either a ``Poster`` or a ``Participant``.
+If the ``Contest`` is ``Private``, a user can be either a ``Poster`` or a ``Participant`` or neither. All the users who are not specified as ``Participant`` by the ``Poster`` s belong to the category of neither. Such users cannot participate in that ``Contest``.
```
Comment (vishwakftw) [inline docs/source/manual.rst]: Could be rephrased better: users who are neither participants nor posters in a private contest will not be allowed to participate in a contest.

**Comment by vbsinha** [comment]
> and/or

Where should I do these? I'm making some changes according to my judgment. Point out if it's the wrong place.

**Comment by vishwakftw** [comment]
Could we add some primitive screenshots? I can work on them.

**Comment by vishwakftw** [comment]
@vbsinha, the comments can be found here: https://github.com/vbsinha/autojudge/pull/81/commits/e438365d79709249d66a57a5134ae4f9003c5dfa

File: docs/source/manual.rst [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+
+Managing ``Problem``
+--------------------
+
+A ``Contest`` consists of ``Problem`` s. Only a ``Poster`` can add / edit / delete ``Problem`` s.
+
+1. ``Problem`` can be added only before the start time of the ``Contest``. To add a ``Problem`` click add ``ADD PROBLEM`` from the ``Contest`` page, and fill the form.
+
+    - A unique problem code distinctly identifies a ``Problem``.
+    - In case the Compilation Script and Test Script are left empty, the default ones are used.
```
Comment (vishwakftw) [inline docs/source/manual.rst]: Add a point telling that you can download the default ones for reference.

File: docs/source/manual.rst [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+    - Fill the other feilds appropriately.
```
Comment (vishwakftw) [inline docs/source/manual.rst]: feilds -> fields

File: docs/source/manual.rst [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+
+2. In the next page, called the ``Problem`` page, add and manage the test-cases. Public test-cases would be visible to the ``Participant`` s while Private ones won't be. Note that test-case addition / deletion will be allowed only till the start of the ``Contest``.
```
Comment (vishwakftw) [inline docs/source/manual.rst]: until the ``Contest`` begins. (consistent with the next point)

**Comment by vbsinha** [comment]
Nice work with the gifs @vishwakftw

**Pull request merged**